### DISCUSSION TITLE
Async action in a set function

### DISCUSSION THREAD

**Discussion by SeongwoonHong** [pmndrs/zustand#863] [Q&A] [answered]
would it be okay to do something like (instead of having multiple set functions)
```
set(async (store) => {
  store.status = 'loading'
  try {
       const something = await asdf()
       store.status = 'succes'
   } catch (e) {
        ...
     }
})
```

**Comment by dai-shi** [comment, marked as answer]
No, you can't mutate `store`. And, `set` must be sync.
I guess you would be interested in [valtio](https://github.com/pmndrs/valtio) for your use case.
